src/Symfony/Component/Console/Tester/ApplicationTester.php

```diff
@@ -62,10 +58,15 @@ public function __construct(Application $application)
     public function run(array $input, $options = array())
     {
         $this->input = new ArrayInput($input);
+
         if (isset($options['interactive'])) {
             $this->input->setInteractive($options['interactive']);
         }
 
+        if ($this->inputs) {
+            $this->input->setStream(self::createStream($this->inputs));
```

Comment:
[Here is the failure culprit](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Console/Application.php#L787), I suggest to `putenv('SHELL_INTERACTIVE=1')` here and, after `run()`, either destroy it or restore its previous value if any